File: banda/IvanRC-banda-Java/src/main/java/modelsBandaMusicos/Guitarrista.java
package modelsBandaMusicos;

public class Guitarrista extends Musico implements IGuitarrista {

    String nombre;
    Double salario = 1500.00;
    int añosExperiencia;
    tipoGuitarra tipoDeGuitarra;

    public Guitarrista(String nombre, int añosExperiencia, tipoGuitarra tipoDeGuitarra) {
        super(nombre, añosExperiencia);
        this.tipoDeGuitarra = tipoDeGuitarra;
    }

    public tipoGuitarra getTipoDeGuitarra() {
        return tipoDeGuitarra;
    }

    public double recalcularSalario(double salario) {
        return IGuitarrista.super.recalcularSalario(salario);
    }
}

File: banda/MarioBueno_Banda_Kotlin/src/main/kotlin/models/Guitarrista.kt
//Mario Bueno López
//maarioo2525@gmail.com

package models

import interfaces.IGuitarrista
import kotlin.math.round

class Guitarrista (override var nombre: String?, override var experiencia: Int, private var tipoDeGuitarra: String): Musico(nombre, experiencia), IGuitarrista {
    override fun respirar() {
        println("Soy $nombre y estoy respirando.")
    }
    override fun interpretar() {
        println("Soy $nombre, músico profesional durante $experiencia años, y estoy interpretando")
    }
    override fun tocarLaGuitarra() {
        println("Estoy tocando un $tipoDeGuitarra")
    }
    companion object {
        var salario: Double = round(Musico.salario * 1.5)
    }
}